### ISSUE TITLE
Restore Windows installer

### ISSUE THREAD

**Issue by pktiuk** [AntiMicroX/antimicroX#4]
This project requires installer for Windows.

Original [antimicro](https://github.com/AntiMicro/antimicro) was compiled on Windows, but this fork doesn't have this feature yet.
Most of required components are already included in antimicro's repository.

It would be nice to also add portable version of AntiMicroX, it was done [there](https://github.com/AntiMicro/antimicro/blob/master/BuildOptions.md)

Things to do:
- [ ] compile AntimicroX on Windows
- [ ] Add build instructions to README
- [ ] Add building and uploading `exe` to github actions on release

This issue is migration of original issue https://github.com/juliagoda/antimicroX/issues/131 from legacy AntiMicroX repo.

**Comment by gombosg** [comment]
While looking around for #32 I discovered this.

We need to learn what the original `antimicro` used as output library that was dropped in favor of Linux, and need to restore that event handler. See [here](https://github.com/AntiMicro/antimicro/tree/master/src/eventhandlers).

**Comment by pktiuk** [comment]
Maybe we should stick to QT-based input methods, just like [Qt Virtual Keyboard](https://doc.qt.io/qt-5/qtvirtualkeyboard-index.html).  
Maybe there is also a class for controlling mouse movement. :thinking:

**Comment by gombosg** [comment]
Are you sure that you can generate OS input events with that? It just looks like... well, a virtual keyboard that the user can interact with.

**Comment by pktiuk** [comment]
It is virtual keyboard, but we could "borrow" some libraries used by this app :)

**Comment by OlegPopow** [comment]
In accordance with the wishes of the respected developer, I would like to ask about the progress of work on the Windows version. Yes, I've been using Windows to play games for some time, because unfortunately Wine does not fully meet my expectations. Some games work tragically on Wine so I had to apologize with Windows and install it after 17 years. I would be delighted if I could use the same program on two platforms and use the same profiles. Don't treat this as a rush, I just want to know if it makes any sense for me to wait or better to look for an alternative.

**Comment by pktiuk** [comment]
Currently, there is no measurable progress in restoring Windows support, now we are investigating possibilities of using more universal (independent on operating system) qt libraries, but while dealing with other issues we are still thinking about avoiding further breaking app for Windows (like avoiding deletion of default icons used by app, because `QIcon::fromTheme` won't work properly with Windows without defined fallback option).

This is still our top priority, but I am not able to deal with this task without preparations and getting familiar with the architecture of AntiMicroX, which is not as small and simple app as end-users may see it.  
I am going to familiarize with structure of this project by fixing its bugs, cleaning its code/configs/files and (probably) improving logging to easier investigate problems with apps (I think it would be great to use multiple logs level in app, it would make debugging easier, because enabling all of logs gives huge output which is difficult to read)

**Comment by AriaMoradi** [comment]
take a look at https://github.com/Tylemagne/Gopher360 the code might help?

**Comment by gombosg** [comment]
Starting to try and install `antimicrox` on Windows... argh... :)

**Comment by gombosg** [comment]
I set up QT Creator on my Windows, and first tried to build the original antimicro using the build instructions there. It found the SDL DLL, compiled, but failed miserably during linking.
I have 32 bit cmake, QT, SDL, everything (except windows).
Attached logs to see if we find anything.
[cmake.txt](https://github.com/AntiMicroX/antimicrox/files/5559369/cmake.txt)
[build.txt](https://github.com/AntiMicroX/antimicrox/files/5559368/build.txt)

**Comment by gombosg** [comment]
In other words, where can I learn more, see logs about what is happening (and going wrong) during linking?

**Comment by pktiuk** [comment]
At first glance it seems that CMake was not searching for SDL2.
It would be good to have in legacy cmake lines (try to add it somewhere at the beginning):
```cmake
find_package(SDL2 REQUIRED)
include_directories(${SDL2_INCLUDE_DIRS})
```
As an example cmake with sdl check [this_link](https://trenki2.github.io/blog/2017/06/02/using-sdl2-with-cmake/)

I see `find_library(SDL2_LIBRARY SDL2 ${SDL2_LIBRARY_DIR})`, but it is less reliable solution than finding packages.

**Comment by gombosg** [comment]
After a lot of work, I managed to get the `FindSDL2.cmake` script running and using `find_package` for SDL2... to get to the same error. xD
Now that I looked through the 2 cmakelist files, I see about a dozen Windows-specific files in antimicro that have been removed from antimicrox.

**Comment by gombosg** [comment]
Ha! I was only 1 step from doing it, doing this https://github.com/AntiMicro/antimicro/pull/207 helped.
Now we can move on to getting things working for `antimicrox`...

**Comment by pktiuk** [comment]
If it helped we should merge this PR.

**Comment by pktiuk** [comment]
Is your current work on any branch?  
In case if someone else would like to try it.

**Comment by gombosg** [comment]
So far I pushed here: https://github.com/AntiMicro/antimicro/tree/windows-build-fix
I used the link you mentioned to install SDL2. Though I haven't done any Linux testing, all this was for Windows.

**Comment by AriaMoradi** [comment]
Ok, first I was doing some research around #114 and exploring the past issues/commits, and now I am starting to understand why juliagoda became so bitter about things in the end. 
Windows support was neglected and I completely understand why, when you don't have access to a windows box and no help is around this is what happens.

To further eliviate the case first we need windows focused devs and testers. And second consider we should this a "port/back port" not a" restoration effort".

**Comment by gombosg** [comment]
Yes. I need a day or so to just sit down and measure how big of an effort this would be. Fortunately, this day is like that. :)

**Comment by AriaMoradi** [comment]
original comment: https://github.com/AntiMicroX/antimicrox/issues/115#issuecomment-738790277
copying it here so we can discuss it further here and not go off-topic in #115
> Ok this is bad! to restore windows we probably have to bring back this deleted code(and probably much more...)

> And also there's this:16b4af6
> which is a part of 2.25(just a few versions after 2.23 which is made by the original team.)

> Then I think to restore windows It maybe best to start from somewhere between 2.25 and 2.23 and port back new features from what we have now to windows...

**Comment by Yay-mod** [comment]
can you make a version that work on windows subsystem for linux

**Comment by gombosg** [comment]
I can get back to this in late March, or April, sorry. It's in progress and there is some hope. :)

**Comment by Bigoukun** [comment]
Good luck in restoring Windows compatibility. I can't help with the code, but I might be able to test any Windows version you'll be able to make.

**Comment by rayman3003** [comment]
Any progress? I'm desperately waiting for Windows version!

**Comment by patricioparga** [comment]
Hey, just found this open issue. I'll look into getting involved soon. Let me know if you need any testing done on windows

**Comment by Zacharybinx34** [comment]
Any progress for Windows? Desperate to have this working again!

**Comment by pktiuk** [comment]
Sorry guys, but we haven't made any significant progress since then.    
Everything done in this matter is (and will be) reported here. If you don't see any updates here you can safely assume there is no progress.

**Comment by NeverEatYellowSwissSnow** [comment]
> 
> 
> Any progress for Windows? Desperate to have this working again!

+1

**Comment by pktiuk** [comment]
First step is made, app compiles on Windows and can be launched, but this task is still far from being done.

Anyway, does anybody have experience with configuring Github Actions for Windows?

**Comment by pktiuk** [comment]
Everything should now work fine.

**Comment by NeverEatYellowSwissSnow** [comment]
Just downloaded _**antimicrox-3.2.0-AMD64.exe**_ but when starting it on Windows 10 nothing happens. I also tried to run it as administrator but still: Nothing happens.
I'm on Windows 10 Version 21H1 (Build 19043.1320)
Any hints?
Thank you!

**Comment by FrankScarlet** [comment]
> Just downloaded _**antimicrox-3.2.0-AMD64.exe**_ but when starting it on Windows 10 nothing happens. I also tried to run it as administrator but still: Nothing happens. I'm on Windows 10 Version 21H1 (Build 19043.1320) Any hints? Thank you!
> 
> EDIT: Got it working running it in Windows 7 compatibility mode.

I'm on Windows 10 Version 21H1 (Build 19043.1348). **antimicrox-3.2.0-AMD64.exe** works without administrator setting.
Also it might be a coincidence that I got **Controller Companion** working running it in Windows7 compatibility mode.
lol
